Meisterplan Connector Update: Added new Allocation Comment endpoints

itd-sk opened the pull request:
- added endpoints for Get One and GET List of allocation comments
- There are a lot of changes in the swagger API definition file because of re-sorting. We internally switched the definition to Open API 3. By converting back to swagger 2, the definitions and end points are displayed in a different order.

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

itd-sk committed:
Add Allocation Comment endpoints

- added endpoints for Get One and GET List of allocation comments

vmanoharas commented on certified-connectors/Meisterplan/apiDefinition.swagger.json:
```diff
@@ -1094,44 +1280,61 @@
+          {
+            "in": "query",
+            "name": "filter",
+            "description": "Stringified JSON object with a subset of properties of the expected object. Selects all objects that exactly match the specified properties. Projects are searchable by projectKey, name, externalId, programId, portfolioId, status, manager and custom fields. Manager fields can be filtered by a resource ID, a resource key. The filter manager = null will yield all projects ...
```
please keep all parameters descriptions short to fit in the text box. please see our [docs](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters).

itd-sk replied:
Thanks, shortened all filter descriptions

vmanoharas commented on certified-connectors/Meisterplan/apiDefinition.swagger.json:
```diff
@@ -1521,49 +1728,54 @@
+          "Priorities"
+        ],
+        "summary": "Update Priorities",
+        "description": "This endpoint updates the priorities of projects and programs. Please consider that updating the priority of a project that is within a program results in unassigning the project from its program.To update the priorities of your projects and programs, list their IDs in the desired order of priority for each rank category. If there are projects or programs in Meisterplan fo ...
```
please keep all parameters descriptions short to fit in the text box. please see our [docs](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters).

itd-sk replied:
Thanks, shortened description

vmanoharas reviewed (commented):
Hello @itd-sk, 

Can you please resolve review comments?

And please check the below validation errors and kindly confirm that the existing flows will not fail?


##[error]The input field has been changed from optional to required. ValuePath: operation/CreateOrUpdateAllocation/parameter/allocationCreateOrUpdateRequest/schema/allocatedEntity/id/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/CreateOrUpdateAllocation/parameter/allocationCreateOrUpdateRequest/schema/segments/items/finish/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/CreateOrUpdateAllocation/parameter/allocationCreateOrUpdateRequest/schema/segments/items/start/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/UpdateAllocation/parameter/allocationUpdateRequest/schema/segments/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/UpdateAllocation/parameter/allocationUpdateRequest/schema/segments/items/finish/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/UpdateAllocation/parameter/allocationUpdateRequest/schema/segments/items/start/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/ReplaceFinancialActuals/parameter/payload/schema/items/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/ReplaceFinancials/parameter/payload/schema/items/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/ReplaceFinancials/parameter/payload/schema/items/items/timing/on/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/CreateFinancials/parameter/payload/schema/timing/on/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/UpdateFinancials/parameter/payload/schema/timing/on/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/ReplaceMilestones/parameter/payload/schema/items/required Source: NewSwagger
##[error]The output field has been changed from required to optional. Existing flows may rely on this field as required, they may be broken. ValuePath: operation/GetAllRoleCapacities/responses/200/schema/items/items/segments/items/fte/required Source: NewSwagger
##[error]The output field has been changed from required to optional. Existing flows may rely on this field as required, they may be broken. ValuePath: operation/GetRoleCapacities/responses/200/schema/items/items/fte/required Source: NewSwagger
##[error]The input field has been changed from optional to required. ValuePath: operation/UpdateRoleCapacities/parameter/payload/schema/segments/required Source: NewSwagger
##[error]The output field has been changed from required to optional. Existing flows may rely on this field as required, they may be broken. ValuePath: operation/UpdateRoleCapacities/responses/200/schema/items/items/fte/required Source: NewSwagger
##[error]The operation has been deleted. This change breaks existing flows. ValuePath: operation/GetAllUsers_1 Source: NewSwagger
##[error]The operation has been deleted. This change breaks existing flows. ValuePath: operation/GetUser_1 Source: NewSwagger

itd-sk committed:
Fix length of description fields

- shorten filter and priorities operation descriptions
- change capacity segment fte response field from optional to required
- rename operation id GetAllUsers to GetAllUsers_1
- rename operation id GetUser to GetUser_1

itd-sk commented:
We reviewed the validation errors. The first 12 errors were intended because these parameters were falsely marked as optional before. The last 6 bugs were not intended and should be fixed with the last commit. Many thanks for pointing out the errors.

vmanoharas commented:
Hello @itd-sk, 

Thank you very much for addressing the errors. 

However, GetAllUsers and GetUser methods are deleted, this may break the existing flow.  Can you please confirm? 

##[error]The operation has been deleted. This change breaks existing flows. ValuePath: operation/GetAllUsers Source: NewSwagger
##[error]The operation has been deleted. This change breaks existing flows. ValuePath: operation/GetUser Source: NewSwagger

itd-sk commented:
Hello @vmanoharas , on the dev branch I can only find the operations GetAllUsers_1 and GetUser_1 and no operation with ids "GetAllUsers"  and "GetUser". So I am wondering why the validation check failed with these two errors. Is the validation task  comparing the swagger file with the swagger file of the commit before where I falsely named  the operations GetAllUsers / GetUser? If this is the case then these two error can be ignored. Shouldn't the validation task "Swagger and Change Validation" compare the new swagger file with the version in the "dev" branch?

vmanoharas commented:
> Hello @vmanoharas , on the dev branch I can only find the operations GetAllUsers_1 and GetUser_1 and no operation with ids "GetAllUsers" and "GetUser". So I am wondering why the validation check failed with these two errors. Is the validation task comparing the swagger file with the swagger file of the commit before where I falsely named the operations GetAllUsers / GetUser? If this is the case then these two error can be ignored. Shouldn't the validation task "Swagger and Change Validation" compare the new swagger file with the version in the "dev" branch?

Hello @itd-sk,  thank you very much for your quick response.  Yes validation checker is taking from old swagger file. Please ignore this error then.

vmanoharas reviewed (approved):
Hello @itd-sk 
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.
